client/src/components/SearchableDropdownField/SearchableDropdownField.js

```diff
@@ -133,6 +133,15 @@ const SearchableDropdownField = ({
     }
   }
 
+  if (Number.isInteger(parseInt(val, 10))) {
+    options.forEach(option => {
+      // eslint-disable-next-line eqeqeq
+      if (option.value == value) {
+        val = option;
+      }
+    });
+  }
+
```

Comment:
`react-select` expect to get value as an array of objects of ValueType in the following format `{ label: 'label', value: 'value' }`. See discussion [here](https://github.com/JedWatson/react-select/issues/2674#issuecomment-401318026).
`FormBuilder.buildComponent()` set `onChange` props as function with 2 arguments (event and payload), but handle payload value first and redux-form set this payload value in the state. [See](https://github.com/silverstripe/silverstripe-admin/blob/a3aa41cea4c4df82050eef65ad5efcfae7bfde69/client/src/components/FormBuilder/FormBuilder.js#L124-L156)
After we render `SearchableDropdownField` again we get integer value from state instead of object. So, if we pass only value that is returned back from `react-select` ([See](https://github.com/silverstripe/silverstripe-admin/blob/a3aa41cea4c4df82050eef65ad5efcfae7bfde69/client/src/components/FormBuilder/FormBuilder.js#L124-L156)) then we process it and store it as an `Object.toString()` and get `[object.Object]` in view. If we pass two params then we get integer or any other value, but not object, after processing.